homeassistant/helpers/device_registry.py

```diff
@@ -356,26 +356,12 @@ def async_clear_area_id(self, area_id: str) -> None:
 
 
 @bind_hass
+@cache_per_instance(DATA_REGISTRY)
```

Comment:
you are essentially creating singletons no? would singleton be a better name for the annotation?

Reply:
Good idea.

Reply:
Done.